lib/AST/AlignTokens.cpp

```diff
@@ -854,12 +768,15 @@ static const std::hash<std::string_view> kHasher;
 // Strip off leading and trailing underscores, then hash. This is to deal with
 // things like `asm` vs. `__asm`.
 static std::string_view HashableData(std::string_view view) {
+  auto size = view.size();
   while (!view.empty() && view.front() == '_') {
-    view = view.substr(1u, view.size() - 1u);
+    --size;
+    view = view.substr(1u, size);
   }
 
   while (!view.empty() && view.back() == '_') {
-    view = view.substr(0, view.size() - 1u);
+    --size;
+    view = view.substr(0, size);
```

Comment:
Extract as generic helpers or use `llvm::StringRef`. Also double check `std::string_view` does not have api for this.